### PR TITLE
✨ Add ISCN-dropdown menu for selecting diff. badge styles

### PR DESCRIPTION
**Pull request by shellyYG**

### CODE REVIEW

File: likecoin/admin/views/components.php [hunk 1]
```diff
@@ -194,3 +194,32 @@ function likecoin_add_generic_checkbox( $args ) {
...
+	>
+	<option value="light"
+		<?php echo isset( $option[ $args['label_for'] ] ) ? ( selected( $option[ $args['label_for'] ], 'light', false ) ) : ( '' ); ?>>
+		<?php esc_html_e( 'Light Mode', LC_PLUGIN_SLUG ); ?>
+	</option>
+	<option value="dark"
+		<?php echo isset( $option[ $args['label_for'] ] ) ? ( selected( $option[ $args['label_for'] ], 'dark', false ) ) : ( '' ); ?>>
+		<?php esc_html_e( 'Dark Mode', LC_PLUGIN_SLUG ); ?>
+	</option>
+	<option value="hidden"
```
Comment (williamchong) [inline likecoin/admin/views/components.php:218]: I think `none` is a better name since it is not really `hidden`. It is actually disabled.

File: likecoin/includes/constant/options.php [hunk 2]
```diff
@@ -57,7 +57,7 @@
 define( 'LC_OPTION_SITE_MATTERS_AUTO_PUBLISH', 'site_matters_auto_publish' );
 define( 'LC_OPTION_SITE_MATTERS_ADD_FOOTER_LINK', 'site_matters_add_footer_link' );
 
-define( 'LC_OPTION_SITE_ADD_ISCN_BADGE', 'site_add_iscn_badge' );
+define( 'LC_OPTION_SITE_ISCN_BADGE_DROPDOWN_STYLE', 'iscn_drop_down_style' );
```
Comment (williamchong) [inline likecoin/includes/constant/options.php:60]: `DROPDOWN` should not be in the name of this option, since dropdown is just one of the possible UI to represent this option. `ISCN_BADGE_STYLE_OPTION` might be better

File: likecoin/public/iscn-badge.php [hunk 3]
```diff
@@ -31,17 +31,34 @@
...
+				break;
+		}
+	}
+
+	if ( strlen( $iscn_hash ) > 0 && 'no' !== $show_iscn_badge ) {
 		$widget_code = '<figure class="likecoin-iscn-badge">' .
 		'<a href="https://like.co/in/tx/iscn/dev/' . $iscn_hash . '" target="_blank" rel="noopener">' .
 		'<img ' .
-		'src="https://static.like.co/badge/iscn/dev/' . $iscn_hash . '">' .
+		'src="https://static.like.co/badge/iscn/dev/' . $iscn_hash . 'IsDarkBadge=' . $is_dark_badge . '"' .
```
Comment (williamchong) [inline likecoin/public/iscn-badge.php:60]: This url is not correctly formatted, querystring needs a `?`, and also `dark` would be better than `IsDarkBadge`.
Is this related to changes in the iscn badge function?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/public/iscn-badge.php [hunk 4]
```diff
@@ -31,17 +31,34 @@
...
-	$option                = get_option( LC_PUBLISH_OPTION_NAME );
-	$should_add_iscn_badge = checked( $option[ LC_OPTION_SITE_ADD_ISCN_BADGE ], 1, false );
+	$post_id   = $post->ID;
+	$iscn_info = get_post_meta( $post_id, LC_ISCN_DEV_INFO, true );
+	$iscn_hash = isset( $iscn_info['iscn_hash'] ) ? $iscn_info['iscn_hash'] : null;
+	$option    = get_option( LC_PUBLISH_OPTION_NAME );
 
-	if ( strlen( $iscn_hash ) > 0 && $should_add_iscn_badge ) {
+	$is_dark_badge   = 0; // default is showing white badge.
+	$show_iscn_badge = 'yes';
```
Comment (williamchong) [inline likecoin/public/iscn-badge.php:40]: Just use a boolean True/False....

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/admin/views/components.php (hunk 1 above)
Comment (shellyYG) [reply likecoin/admin/views/components.php:218]: gotcha, yes makes sense. done~

File: likecoin/public/iscn-badge.php (hunk 3 above)
Comment (shellyYG) [reply likecoin/public/iscn-badge.php:60]: yes, related :) just changed

File: likecoin/public/iscn-badge.php (hunk 4 above)
Comment (shellyYG) [reply likecoin/public/iscn-badge.php:40]: true...lol

Comment (shellyYG) [reply likecoin/public/iscn-badge.php:40]: not native enough yet for programming language xD

File: likecoin/public/iscn-badge.php [hunk 5]
```diff
@@ -37,27 +37,27 @@ function likecoin_add_iscn_badge( $post ) {
...
-			case 'hidden':
-				$show_iscn_badge = 'no';
+			case 'none':
+				$show_iscn_badge = false;
 				break;
 		}
 	}
 
-	if ( strlen( $iscn_hash ) > 0 && 'no' !== $show_iscn_badge ) {
+	if ( strlen( $iscn_hash ) > 0 && false !== $show_iscn_badge ) {
```
Comment (williamchong) [inline likecoin/public/iscn-badge.php:56]: Just `$show_iscn_badge` would do? Since it is a boolean

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/public/iscn-badge.php [hunk 6]
```diff
@@ -31,17 +31,34 @@
...
+		$iscn_badge_style_option = $option[ LC_OPTION_ISCN_BADGE_STYLE_OPTION ];
+		switch ( $iscn_badge_style_option ) {
+			case 'dark':
+				$is_dark_badge = 1;
+				break;
+			case 'white':
+				break;
+			case 'none':
+				$show_iscn_badge = false;
+				break;
```
Comment (williamchong) [inline likecoin/public/iscn-badge.php:52]: Default case is `true` seems weird, I suggest default should be false, and setting `show_iscn_badge` to `true` to when case is `dark` and `white`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/public/iscn-badge.php (hunk 6 above)
Comment (shellyYG) [reply likecoin/public/iscn-badge.php:52]: ok!

File: likecoin/public/iscn-badge.php (hunk 5 above)
Comment (shellyYG) [reply likecoin/public/iscn-badge.php:56]: gotcha! yes, tested and it works

File: likecoin/admin/views/components.php [hunk 7]
```diff
@@ -194,3 +194,32 @@ function likecoin_add_generic_checkbox( $args ) {
...
+		<?php esc_html_e( 'Light Mode', LC_PLUGIN_SLUG ); ?>
+	</option>
+	<option value="dark"
+		<?php echo isset( $option[ $args['label_for'] ] ) ? ( selected( $option[ $args['label_for'] ], 'dark', false ) ) : ( '' ); ?>>
+		<?php esc_html_e( 'Dark Mode', LC_PLUGIN_SLUG ); ?>
+	</option>
+	<option value="none"
+		<?php echo isset( $option[ $args['label_for'] ] ) ? ( selected( $option[ $args['label_for'] ], 'none', false ) ) : ( '' ); ?>>
+		<?php esc_html_e( 'None', LC_PLUGIN_SLUG ); ?>
+	</option>
```
Comment (nwingt) [inline likecoin/admin/views/components.php:221]: Consider using for-loop to render this

**Review by nwingt — CHANGES_REQUESTED** [review]

File: likecoin/public/iscn-badge.php [hunk 8]
```diff
@@ -31,17 +31,35 @@
...
+		}
+	}
+
+	if ( strlen( $iscn_hash ) > 0 && $show_iscn_badge ) {
 		$widget_code = '<figure class="likecoin-iscn-badge">' .
 		'<a href="https://like.co/in/tx/iscn/dev/' . $iscn_hash . '" target="_blank" rel="noopener">' .
 		'<img ' .
-		'src="https://static.like.co/badge/iscn/dev/' . $iscn_hash . '">' .
+		'src="https://static.like.co/badge/iscn/dev/' . $iscn_hash . '?dark=' . $is_dark_badge . '"' .
```
Comment (williamchong) [inline likecoin/public/iscn-badge.php:61]: should add `.svg` after `$iscn_hash`

**Review by williamchong — CHANGES_REQUESTED** [review]

File: likecoin/public/iscn-badge.php (hunk 8 above)
Comment (shellyYG) [reply likecoin/public/iscn-badge.php:61]: ok!

File: likecoin/admin/views/components.php (hunk 7 above)
Comment (shellyYG) [reply likecoin/admin/views/components.php:221]: done~

Comment (shellyYG) [reply likecoin/admin/views/components.php]: seems failing with esc_html__e only accept single string...will keep this as-is for now and come back later

**Review by williamchong — APPROVED** [review]
